assets/js/Sign.jsx

```diff
@@ -1,18 +1,31 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-function Sign({ signId, lineOne, lineTwo }) {
+function Sign({
+  signId, lineOne, lineOneDuration, lineTwo, lineTwoDuration, currentTime,
+}) {
+  const remainingTimeOne = Date.parse(lineOneDuration) - currentTime;
```

Comment:
you could extract this into a separate function in this file, that takes duration and currentTime and returns a boolean, true if it's greater than 0, false otherwise.

and then in the return JSX:

{ remainingTimeOne ? lineOne : null }
...
{ remainingTimeTwo ? lineTwo : null }

Reply:
good idea